pkg/migration/controllers/migrationschedule.go

```diff
@@ -0,0 +1,306 @@
+	migrationName := m.formatMigrationName(migrationSchedule, policyType)
+	if migrationSchedule.Status.Items == nil {
+		migrationSchedule.Status.Items = make(map[stork_api.SchedulePolicyType][]*stork_api.ScheduledMigrationStatus)
+	}
+	if migrationSchedule.Status.Items[policyType] == nil {
+		migrationSchedule.Status.Items[policyType] = make([]*stork_api.ScheduledMigrationStatus, 0)
+	}
+	migrationSchedule.Status.Items[policyType] = append(migrationSchedule.Status.Items[policyType],
+		&stork_api.ScheduledMigrationStatus{
+			Name:              migrationName,
+			CreationTimestamp: meta.Now(),
+			Status:            stork_api.MigrationStatusPending,
+		})
+	err := sdk.Update(migrationSchedule)
+	if err != nil {
+		return err
+	}
+
+	migration := &stork_api.Migration{
+		ObjectMeta: meta.ObjectMeta{
+			Name:      migrationName,
+			Namespace: migrationSchedule.Namespace,
+			OwnerReferences: []meta.OwnerReference{
+				{
+					Name:       migrationSchedule.Name,
+					UID:        migrationSchedule.UID,
+					Kind:       migrationSchedule.GetObjectKind().GroupVersionKind().Kind,
+					APIVersion: migrationSchedule.GetObjectKind().GroupVersionKind().GroupVersion().String(),
+				},
+			},
+		},
+		Spec: migrationSchedule.Spec.Template.Spec,
+	}
+	log.MigrationScheduleLog(migrationSchedule).Infof("Starting migration %v", migrationName)
+	_, err = k8s.Instance().CreateMigration(migration)
+	return err
+}
+
+func (m *MigrationScheduleController) pruneMigrations(migrationSchedule *stork_api.MigrationSchedule) error {
+	for policyType, policyMigration := range migrationSchedule.Status.Items {
+		// Keep only one successful migration status and all failed migrations
+		// until there is a successful one
+		numMigrations := len(policyMigration)
+		deleteBefore := 0
+		if numMigrations > 1 {
+			// Start from the end and find the last successful migration
+			for i := range policyMigration {
+				if policyMigration[(numMigrations-i-1)].Status == stork_api.MigrationStatusSuccessful {
+					deleteBefore = numMigrations - i - 1
+					break
+				}
+			}
+			for i := 0; i < deleteBefore; i++ {
+				err := k8s.Instance().DeleteMigration(policyMigration[i].Name, migrationSchedule.Namespace)
+				if err != nil {
+					log.MigrationScheduleLog(migrationSchedule).Warnf("Error deleting %v: %v", policyMigration[i].Name, err)
+				}
+			}
+			migrationSchedule.Status.Items[policyType] = policyMigration[deleteBefore:]
+		}
+	}
+	return sdk.Update(migrationSchedule)
+}
+
+func (m *MigrationScheduleController) deleteMigrations(migrationSchedule *stork_api.MigrationSchedule) error {
+	for _, policyMigration := range migrationSchedule.Status.Items {
+		for _, migration := range policyMigration {
+			err := k8s.Instance().DeleteMigration(migration.Name, migrationSchedule.Namespace)
+			if err != nil {
+				log.MigrationScheduleLog(migrationSchedule).Warnf("Error deleting %v: %v", migration.Name, err)
```

Comment:
save atleast the last non-nil error and return at the end.

Reply:
Returning the last error after Ignoring "NotFound" errors